libc-top-half/stub-pthreads/stub-pthreads.c

```diff
@@ -0,0 +1,78 @@
+#include "pthread_impl.h"
+
+static void dummy_0()
+{
+}
+weak_alias(dummy_0, __acquire_ptc);
+weak_alias(dummy_0, __release_ptc);
+weak_alias(dummy_0, __pthread_tsd_run_dtors);
+
+int __pthread_create(pthread_t *restrict res, const pthread_attr_t *restrict attrp, void *(*entry)(void *), void *restrict arg)
+{
+	/*
+		"The system lacked the necessary resources to create another thread,
+		or the system-imposed limit on the total number of threads in a process
+		{PTHREAD_THREADS_MAX} would be exceeded."
+	*/
+	return EAGAIN;
+}
```

Comment:
(I don't have write permission for this repo)

It feels to me like pthread_create, pthread_detach and pthread_join should be kept in a separate archive just like mmap (libwasi-emulated-mman.a) and the signal handler stubs (libwasi-emulated-signal.a) pthread_once and pthread_exit are fine to keep in the main libc.a given that (unlike pthread_create and co) they are actual implementations behaving as they should according to the POSIX specification, albeit only working correctly on single threaded targets, but they are only used on single threaded targets anyway.

Reply:
I would be happy to test a proposal implementing this as a solution. @sbc100 @yamt Do you think we could use that approach?